### PR TITLE
Deliverable

### PR DESCRIPTION
**Pull request by gabrielaRazo**
All test unit cases done plus mock.

### CODE REVIEW

File: src/Components/Dashboard/dashboard.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    const [unexpectedError, setUnexpectedError] = useState(false)
+    const [date, setDate] = useState(moment(new Date()).format("YYYY-MM-DD"))
+
+
+    const fetchMyAPI = useCallback(async () => {
+        if(moment(date).isAfter(moment(new Date()).format("YYYY-MM-DD"))===true || moment('1995-06-16').isSameOrAfter(date)===true ){
+            setErrorMsg(true)
+        }else{
+            const apiKey = 'wOjD8BmGH4ZViJyv2OpeZlKfSaA24XkEcvmyipkF';
+            let response = await fetch(`https://api.nasa.gov/planetary/apod?api_key=${apiKey}&start_date=${date}&end_date=${date}`)
```
Comment (gusgarciaw) [inline src/Components/Dashboard/dashboard.js]: Your app is looking good with the current changes but it's a good practice to abstract the fetching logic and states related to a custom hook so with this approach the components are more stateless and easier to apply tests.
Feel free to not implement the changes now, but it's a best practice to have custom hooks.
[Article How to fetch data with React Hooks](https://www.robinwieruch.de/react-hooks-fetch-data/)
[CodeSandbox useFetch](https://codesandbox.io/s/usereducer-example-71pbj8?file=/src/App.js)

File: src/Components/Dashboard/dashboard-test.test.js [hunk 2]
```diff
@@ -15,16 +16,35 @@ describe('Dasboard', () => {
...
+                  error: "error"
+                })
+              );
+            })
+        );
+        setup();
+        const msgUnexError = await screen.getByText(/There was an error, please try again./i);
+        expect(msgUnexError).toBeInTheDocument();
+    })
+    it('the app should show a message from the API response when the user selects an invalid date value ', async () => {
```
Comment (gusgarciaw) [inline src/Components/Dashboard/dashboard-test.test.js]: Good Job 👍

File: src/components/dashboard/dashboard-test.test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+describe('Dashboard', () => {
+    it('the app should show the Picture of the Day', async () => {
+        setup()
+        const dayOfToday = moment(new Date()).format("YYYY-MM-DD");
+        const imgResult = await screen.findByRole('img', {altText: dayOfToday})
+        expect(imgResult).toBeInTheDocument();
+        
+    })
+    it('the app should show the picture of the day for the given date when the user selects a date', async () => {
```
Comment (gusgarciaw) [inline src/components/dashboard/dashboard-test.test.js:17]: Good Job 👍

File: src/components/dashboard/dashboard-test.test.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        setup()
+        const input = await screen.findByLabelText(/fecha/i)
+        fireEvent.change(input, {target: {value: '2021-02-12'}})
+        const regex = /^\d{4}-\d{2}-\d{2}$/;
+        const imgResult = await screen.findByRole('img', {altText: '2021-02-12'})
+        expect(input.value).toMatch(regex)
+        expect(imgResult).toBeInTheDocument();
+    })
+    it('the app should show a message: "There was an error, please try again." when the app fetches the API, and there is an unexpected error,', async () => {
```
Comment (gusgarciaw) [inline src/components/dashboard/dashboard-test.test.js:26]: Good Job 👍

**Review by gusgarciaw — COMMENTED** [review]
Hi Gaby 👋! I made a couple of comments about code refactoring but in general,  good job, your app and the tests are awesome.

**Comment by gabrielaRazo** [comment]
Thanks for the feedback. I am glad that I didn't have major errors. I'm
looking forward to improve my code.

Greetings.

**Pull request merged**